datadog_lambda/tracing.py

```diff
@@ -411,6 +411,9 @@ def is_legacy_lambda_step_function(event):
     """
     Check if the event is a step function that called a legacy lambda
     """
+    if not isinstance(event, dict):
```

Comment:
do we have other code that assumes that this is a dict in the wrapper flow that will trip over this problem?

also, should we look at the contents of the list if it is a list? or is that impossible for a legacy lambda step function?

Reply:
I dont' think we need to look through the whole list. If it's not a dict, we know right away that it's not a step function event.

As for the isinstance, we do have other checks for if the event is a dict. For example in determining trace propagation we skip altogether if it's not a dict. Best answer could even be surrounding the whole thing with a try/except.